File: Beecrowd/1068/1068.cpp
#include <iostream>
#include <vector>
#include <string>
using namespace std;

bool parenthesis(string exp){
    vector<char>stack;
    for(int i = 0; i < exp.length(); i++){
        if(exp[i] == '('){
            stack.push_back('(');
        }
        else if(exp[i] == ')'){
            if(stack.empty()){
                return false;
            }else{
                stack.pop_back();
            }
        }
    }
    return stack.empty();
}

int main(){
    string exp;
    while(getline(cin, exp)){
        if(parenthesis(exp)){
            cout << "correct" << endl;
        }else{
            cout << "incorrect" << endl;
        }
    }

    return 0;
}

File: Beecrowd/1068/1068.py
# Problem statement: https://www.beecrowd.com.br/judge/en/problems/view/1068
while True:
    try:
        exp = input()
        possible = True
        stack = []
        for i in exp:
            if i == '(':
                stack.append('(')
            elif i == ')':
                if stack.count('(') == 0:
                    possible = False
                else:
                    stack.pop()
        
        if possible == True and stack.count('(') == 0:
            print("correct")
        else:
            print("incorrect")
    except EOFError:
        break;

